### ISSUE TITLE
Getting didChangeWatchedFiles events for git: URIs

### ISSUE THREAD

**Issue by davidanthoff** [microsoft/vscode#91242]
We are registering for file change events with our language client like this:
```
LanguageClientOptions = {
        documentSelector: ['julia', 'juliamarkdown'],
        synchronize: {
            fileEvents: vscode.workspace.createFileSystemWatcher('**/*.{jl,jmd}')
        }
    }
```
In our language server we are now receiving `workspace/didChangeWatchedFiles` notifications for URIs that start with `git:`.

Two questions:
- why are we receiving these at all? I would have assumed that the file system watcher we create there is watching for file changes on disc with the normal `file:` schema.
- how are we supposed to handle these notifications? For now we are just ignoring anything that doesn't start with `file:`, but it is unclear to me whether that is the right approach. But then, I also don't see what we could meaningfully do otherwise?

**Comment by joaomoreno** [comment]
> I would have assumed that the file system watcher we create there is watching for file changes on disc with the normal file: schema.

Your glob says otherwise... you should probably pass a glob `file://**/*.{jl,jmd}`. `git:` isn't the only schema out there, you'll get events for other uris as well.

Ignoring everything but `file:` is the right approach. 👍

**Comment by davidanthoff** [comment]
Ok, thanks, that helps!

I find it a bit weird, though :) Things with a schema like `untitled:` or `git:` do not exist on the file system, right? So why would the file system watcher send notifications about them? I understand that I'll get `didOpen` messages in our language server for these, but getting them from the file system watcher seems strange to me.

The documentation on this is also not great. It says `A glob pattern that filters the file events on their absolute path must be provided.` Note that it talks about _path_, not URI (which seems to be what is really going on here).

**Comment by joaomoreno** [comment]
Virtual file systems are still file systems, they can still emit events. ;)

cc @jrieken on docs

**Comment by davidanthoff** [comment]
The docs [here](https://code.visualstudio.com/api/references/vscode-api#FileSystemWatcher) should probably also be changed, right now it says "A file system watcher notifies about changes to files and folders on disk."

**Comment by bpasero** [comment]
@jrieken thoughts? should we be blocking the events for `git` URIs? Seems odd though because maybe some other extension is utilizing the fact that git is now registering a proper file system provider (maybe GitLens @eamodio ?).

Anyhow, updating the docs probably makes sense.

**Comment by jrieken** [comment]
Yeah, change the docs is the right thing to do. All file systems (FileSystemProvider-API) fire events that are observable like this, document content provider don't

**Comment by davidanthoff** [comment]
I don't know what the policy around breaking changes is for VS Code, but this did break our extension. We relied on the documented behavior, and I think in previous versions of VS Code the only notifications that were sent might have been `file:`? In any case, we just recently started seeing crash reports for the Julia extension where our language server crashed because we didn't handle these non `file:` cases properly. Obviously super easily fixed on our end, but if you have a policy of not breaking shipping versions of extensions, then maybe this should be classified as breaking (but might also be a border line case).
